homeassistant/components/media_player/channels.py

```diff
@@ -0,0 +1,303 @@
+        else:
+            self.channel_number = None
+            self.channel_name = None
+            self.channel_image_url = None
+
+        if np_hash:
+            self.now_playing_title = np_hash.get('title')
+            self.now_playing_episode_title = np_hash.get('episode_title')
+            self.now_playing_season_number = np_hash.get('season_number')
+            self.now_playing_episode_number = np_hash.get('episode_number')
+            self.now_playing_summary = np_hash.get('summary')
+            self.now_playing_image_url = np_hash.get('image_url')
+        else:
+            self.now_playing_title = None
+            self.now_playing_episode_title = None
+            self.now_playing_season_number = None
+            self.now_playing_episode_number = None
+            self.now_playing_summary = None
+            self.now_playing_image_url = None
+
+    @property
+    def name(self):
+        """Return the name of the player."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the player."""
+        if self.status == 'stopped':
+            return STATE_IDLE
+
+        if self.status == 'paused':
+            return STATE_PAUSED
+
+        if self.status == 'playing':
+            return STATE_PLAYING
+
+        return None
+
+    def update(self):
+        """Retrieve latest state."""
+        self.update_favorite_channels()
+        self.update_state(self.client.status())
+
+    @property
+    def source_list(self):
+        """List of favorite channels."""
+        sources = [channel['name'] for channel in self.favorite_channels]
+        return sources
+
+    @property
+    def is_volume_muted(self):
+        """Boolean if volume is currently muted."""
+        return self.muted
+
+    @property
+    def media_content_id(self):
+        """Content ID of current playing channel."""
+        return self.channel_number
+
+    @property
+    def media_content_type(self):
+        """Content type of current playing media."""
+        return MEDIA_TYPE_CHANNEL
+
+    @property
+    def media_image_url(self):
+        """Image url of current playing media."""
+        if self.now_playing_image_url:
+            return self.now_playing_image_url
+        elif self.channel_image_url:
+            return self.channel_image_url
+
+        return 'https://getchannels.com/assets/img/icon-1024.png'
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        if self.state:
+            return self.now_playing_title
+
+        return None
+
+    @property
+    def supported_features(self):
+        """Flag of media commands that are supported."""
+        return FEATURE_SUPPORT
+
+    def mute_volume(self, mute):
+        """Mute (true) or unmute (false) player."""
+        if mute != self.muted:
+            response = self.client.toggle_muted()
+            self.update_state(response)
```

Comment:
Since this is a polling entity, `update` will be called directly after the service call. So is this needed?

Reply:
OK great, I'm glad you brought this up. So, every response from the API returns the actual updated state. Because of that, I was attempting to just update the state immediately after so there was no lag with polling.

But what I found was the UI isn't actually updating immediately like I would expect with this code. The UI still only updates with using the normal poll period.

How should I be updating the UI immediately so it feels responsive?

For example, when you click pause to pause it, it doesn't update the play/pause button immediately. Thanks!

Reply:
All services from the media_player base component should take care of calling update and updating the state. For the services that you've implemented in this platform, you named to add that to the service handler. Look at the source for the service handler in the base media_player component, for an example. You will not use the async methods here though.

I'm not sure why it doesn't seem to be responsive when calling the service implemented in the base component.